core/state/statedb.go

```diff
@@ -751,7 +750,9 @@ func (s *StateDB) Copy() *StateDB {
 		state.stateObjectsDirty[addr] = struct{}{}
 	}
 	// Deep copy the destruction markers.
-	state.stateObjectsDestruct = maps.Clone(s.stateObjectsDestruct)
+	for addr, account := range s.stateObjectsDestruct {
+		state.stateObjectsDestruct[addr] = account.Copy()
```

Comment:
The original value of the destructed account is actually immutable. We don't really need to do the deep-copy here.